Sources/MapboxDirections/RouteStep.swift

```diff
@@ -534,7 +543,42 @@ open class RouteStep: Codable {
         }
     }
     
-    public required init(from decoder: Decoder) throws {
+    static func decode(from decoder: Decoder, administrativeRegions: [AdministrativeRegion]) throws -> [RouteStep] {
+        var container = try decoder.unkeyedContainer()
+        
+        var steps = Array<RouteStep>()
+        while !container.isAtEnd {
+            let step = try RouteStep(from: container.superDecoder(), administrativeRegions: administrativeRegions)
+            
+            steps.append(step)
+        }
+        
+        return steps
+    }
+    
+    /// Used to Decode `Intersection.admin_index`
+    private struct AdministrativeAreaIndex: Codable {
+        private enum CodingKeys: String, CodingKey {
+            case administrativeRegionIndex = "admin_index"
+        }
+        
+        var administrativeRegionIndex: Int?
+    }
+
+    /// Used to Decode `Intersection.geometry_index`
+    private struct IntersectionIndex: Codable {
```

Comment:
Since this struct is used only inside `RouteStep` for specific decoding scenario, I would prefer to have it `private`ly nested into `RouteStep`. But indeed some `shape` suffix could make it more descriptable.